### DISCUSSION TITLE
SearchNode CPU cores

### DISCUSSION THREAD

**Discussion by johnununu** [Security-Onion-Solutions/securityonion#7022] [Unsupported Versions] [answered]
Currently I have 4 search nodes running and a manager_search on a robust VM cluster.
I have assigned 16 cores on one virtual cpu and 128G Ram on the search nodes.

Would there be value in splitting the virtual CPUs to 2 CPUs with 8 cores?  I wouldn't think so, but I need to solve a high CPU issue.

I can create custom pipelines and create custom dashboards using the results of the pipelines no problem.
I can filter in the dashboard no problem as well.
If I raw search for anything in kibana, CPU shoots up to 99+% and elasticsearch container shows over 1000% cpu usage from a 'docker stats' command.

![image](https://user-images.githubusercontent.com/87878693/151599388-d6b69d63-de64-4923-a56b-97022820b377.png)

Or is 4 not enough and I need to deploy more search nodes?

I've tried different things to solve this issue.  It's lead to re-builds as well.

**Comment by dougburks** [comment, marked as answer]
> Would there be value in splitting the virtual CPUs to 2 CPUs with 8 cores?

Probably not.

> I can create custom pipelines and create custom dashboards using the results of the pipelines no problem.

What kind of custom pipelines have you created?

> If I raw search for anything in kibana, CPU shoots up to 99+% and elasticsearch container shows over 1000% cpu usage from a 'docker stats' command.

Does the same thing happen if you search in our Hunt interface?
https://docs.securityonion.net/en/2.3/hunt.html

> Or is 4 not enough and I need to deploy more search nodes?

Hard to say without more information:

Are you using spinning disks or SSD?

Are you using cross cluster search or Elastic clustering?

How much data are you ingesting?

Have you checked the logs for additional clues?

Are all the search nodes in the same data center as the manager?

Have you checked Elasticsearch heap size?
https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

Have you checked shard size?
https://docs.securityonion.net/en/2.3/elasticsearch.html#shards

**Reply by johnununu** [reply]
What kind of custom pipelines have you created?
- Cisco WSA and IOS.  Logs were not parsing out the way we needed them to using the built in filebeat modules.

Are you using spinning disks or SSD?
- Some SSD, but mostly HDD.

Are you using cross cluster search or Elastic clustering?
- I did the basic install, cross cluster

How much data are you ingesting?
- Over the last 24hrs, roughly 655,000,000 logs.  Or about 30,000,000/hr.
- We are wanting to add another log source to correlate between sources 

Have you checked the logs for additional clues?
-I've checked the logs on the hardware and the security onion VMs.  Nothing really shows why the cpu spikes during a search.

Are all the search nodes in the same data center as the manager?
- Most are.  There are 2 search nodes in a different datacenter.

Have you checked Elasticsearch heap size?
- esheap size is at 25G on all search nodes
https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

Have you checked shard size?
- Modified the shards.  If the filebeat modules are used, like cisco modules, only one shard is created.  I modified cisco shards to 7 and syslog shards to 10.  No shard is over 40Gb.  Most are under 10Gb.
https://docs.securityonion.net/en/2.3/elasticsearch.html#shards

**Reply by dougburks** [reply]
Next thing I'd check is disk I/O. Also check to see there is a difference in the SSD nodes vs HDD nodes.

**Reply by johnununu** [reply]
I figured it out.
I increased virtual CPU from 16 to 30.  So doubling the CPU, I no longer spike, but hover between 60% and 90%.
I did that in the past and it didn't help, which lead to a rebuild.
Although hind sight, the rebuild was unnecessary.  At the time, I didn't know enabling the cisco ftd and ios filebeat modules meant only 1 shard was going to be enabled.  That clearly wasn't enough, now I have 5 so-cisco shards running.
With only 1 shard and CPU spiking at 99% it left all dashboards erroring out.  Hunt, Alert, all Kibana.
Adding the 5 shards and CPU spiking had no affect on your custom hunt or alert dashboards.  Those were still very useful.  So thank you for all of you and your teams hard work and thoughtful development.

![image](https://user-images.githubusercontent.com/87878693/152551434-6cf4f909-338e-4548-ab28-cacca0ade692.png)

**Comment by johnununu** [comment]
Please mark as answered.

**Reply by dougburks** [reply]
I've marked as answered. Just for future reference, you should be able to do this yourself.
